prov/gni/src/gnix_cntr.c

```diff
@@ -73,23 +73,47 @@ static int __verify_cntr_attr(struct fi_cntr_attr *attr)
 		return -FI_EINVAL;
 	}
 
-	/* TODO: Wait objects are not yet implemented. */
-	if (attr->wait_obj != FI_WAIT_NONE) {
-		return -FI_ENOSYS;
-	}
-
 	switch (attr->wait_obj) {
 	case FI_WAIT_UNSPEC:
 	case FI_WAIT_NONE:
-		break;
 	case FI_WAIT_SET:
+		break;
 	case FI_WAIT_FD:
 	case FI_WAIT_MUTEX_COND:
 	default:
 		GNIX_WARN(FI_LOG_CQ, "wait type: %d unsupported.\n",
 			  attr->wait_obj);
 		return -FI_EINVAL;
-		ret = -FI_ENOSYS;
+		break;
```

Comment:
I prefer leaving in the break regardless of having a return as a defensive coding measure but I can remove it if necessary, the return -FI_EINVAL was unchanged from previous I only removed the extraneous ret = -FI_ENOSYS, I could also update that if deemed necessary.